src/shared/errors/HttpError.ts

```diff
@@ -0,0 +1,11 @@
+export class HttpError extends Error {
+  status: number;
+  data: unknown;
+
+  constructor(status: number, data?: unknown) {
+    super(`HTTP Error ${status}`);
+    this.name = 'HttpError';
+    this.status = status;
+    this.data = data;
```

Comment:
```suggestion
  constructor(public status: number, public data?: unknown) {
    super(`HTTP Error ${status}`);
    this.name = 'HttpError';
```
아마 이렇게 정의하면 자동으로 할당될 것 같아요!
javascript에서는 안 되고, typescript 에서만 가능한 문법이랍니다 ㅎㅎ